sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationExec.scala

```diff
@@ -175,15 +168,7 @@ case class HiveScriptTransformationExec(
         if (outputSerde == null) {
```

Comment:
Seems like we don't need this check every `next()` called, so how about making a specific iterator implementation for the case without serde?

Reply:
> Seems like we don't need this check every `next()` called, so could we avoid this?

Done